src/tcgwars/logic/impl/gen8/SwordShieldPromos.groovy

```diff
@@ -835,6 +875,298 @@ public enum SwordShieldPromos implements LogicCardInfo {
+      return copy (VividVoltage.ARROKUDA_41, this)
+      case DRAGAPULT_VMAX_SWSH97:
+      return copy (VividVoltage.ARROKUDA_41, this)
+      case CROBAT_V_SWSH98:
+      return copy (VividVoltage.ARROKUDA_41, this)
+      case CROBAT_VMAX_SWSH99:
+      return copy (VividVoltage.ARROKUDA_41, this)
+      case VENUSAUR_V_SWSH100:
+      return basic (this, hp:HP220, type:G, retreatCost:3) {
+        weakness R
+        move "Leaf Drain", {
+          text "50 damage. Heal 30 damage from this Pokemon."
+          energyCost G, C
+          attackRequirement {}
+          onAttack {
+            damage 50
+            heal 30, self
+          }
+        }
+        move "Double-Edge", {
+          text "190 damage. This Pokemon also does 30 damage to itself."
+          energyCost G, G, C
+          attackRequirement {}
+          onAttack {
+            damage 190
+            damage 30, self
+          }
+        }
+      }
+      case BLASTOISE_V_SWSH101:
+      return basic (this, hp:HP220, type:W, retreatCost:3) {
+        weakness L
+        move "Water Gun", {
+          text "30 damage."
+          energyCost W
+          attackRequirement {}
+          onAttack {
+            damage 30
+          }
+        }
+        move "Torrential Cannon", {
+          text "200 damage. During your next turn, this Pokémon can’t use Torrential Cannon."
+          energyCost W, W, W
+          attackRequirement {}
+          onAttack {
+            damage 200
+            cantUseAttack(thisMove, self)
+          }
+        }
+      }
+      case VENUSAUR_VMAX_SWSH102:
+      return evolution (this, from: "Venusaur V", hp:HP330, type:G, retreatCost:4) {
+        weakness R
+        move "Forest Storm", {
+          text "30x damage. This attack does 30 damage for each [G] Energy attached to all of your Pokémon."
+          energyCost C, C
+          attackRequirement {}
+          onAttack {
+            my.all.each { damage 30*it.cards.energyCount(G) }
+          }
+        }
+        move "G-Max Bloom", {
+          text "210 damage. Heal 30 damage from this Pokemon."
+          energyCost G, G, C, C
+          attackRequirement {}
+          onAttack {
+            damage 210
+            heal 30, self
+          }
+        }
+      }
+      case BLASTOISE_VMAX_SWSH103:
+      return evolution (this, from: "Blastoise V", hp:HP330, type:W, retreatCost:3) {
+        weakness L
+        move "Grand Falls", {
+          text "120 damage. Search your deck for up to 3 [W] Energy cards and attach them to your Benched Pokémon in any way you like. Then, shuffle your deck."
+          energyCost W, W, W
+          attackRequirement {}
+          onAttack {
+            damage 120
+            attachEnergyFrom(type: W, max: 1, my.deck, my.bench.select())
+            attachEnergyFrom(type: W, max: 1, my.deck, my.bench.select())
+            attachEnergyFrom(type: W, max: 1, my.deck, my.bench.select())
+            shuffleDeck()
+          }
```

Comment:
The attack is missing a requirement for having cards in deck and benched Pokémon, plus the same checks for the energy search itself as an if (in case it gets copied by the opponent's active). 

The effect can be simplified taking Gardevoir&Sylveon-GX (UNB 130)'s "Fairy Song" as a reference. It's better to search all energies at once, then attach them as desired :)